modules/thermo-cycler/thermo-cycler-arduino/gcode.h

```diff
@@ -0,0 +1,56 @@
+#ifndef GCODE_H
+#define GCODE_H
+
+#include "Arduino.h"
+
+#define TOTAL_GCODE_COMMAND_CODES   11
+#define MAX_SERIAL_BUFFER_LENGTH    100
+#define MAX_SERIAL_DIGITS_IN_NUMBER 7
+#define SERIAL_DIGITS_IN_RESPONSE   3
+
+typedef enum{
+  GCODE_NO_CODE = -1,
+  GCODE_GET_LID_STATUS = 0,
+  GCODE_OPEN_LID,
+  GCODE_CLOSE_LID,
+  GCODE_SET_LID_TEMP,
+  GCODE_DEACTIVATE_LID_HEATING,
+  GCODE_SET_PLATE_TEMP,
+  GCODE_GET_PLATE_TEMP,
+  GCODE_SET_RAMP_RATE,
+  GCODE_EDIT_PID_PARAMS,
+  GCODE_PAUSE,
+  GCODE_DEACTIVATE_ALL,
+  GCODE_DFU
+}Gcode_enum;
+
+class Gcode{
```

Comment:
We should have some more comments here describing how the interface to this class is supposed to work.